middleware/decrypt.py

```diff
@@ -0,0 +1,76 @@
+"""Identify and decrypt Crypt4GH keys and files.
+
+Decrypts crypt4GH files given a list of files and places the output in a specified
+directory. If a file is not Crypt4GH-encrypted, it moves the file to the output directory
+without alteration. If a Crypt4GH file is included, the private key associated with that
+file must be provided.
+
+Example:
+    python3 decrypt.py --output-dir /outputs/ file.txt file.c4gh sk.sec pk.pub
+"""
+import shutil
+from argparse import ArgumentParser
+from pathlib import Path
+from crypt4gh.lib import decrypt  # type: ignore
+from crypt4gh.keys import get_private_key  # type: ignore
+
+
+def get_private_keys(file_paths: list[Path]) -> list[bytes]:
+    """Retrieve private keys from a list of files.
+
+    Args:
+        file_paths (list[Path]): A list of file paths.
+
+    Returns:
+        list[bytes]: A list of retrieved private keys as byte objects.
+    """
+    private_keys = []
+    for file_path in file_paths:
+        try:
+            key = get_private_key(Path(file_path), lambda x: '')
+            private_keys.append(key)
+        except ValueError:
+            continue
+    return private_keys
+
+
+def decrypt_files(
+        file_paths: list[Path],
+        private_keys: list[bytes]):
+    """Decrypt files and save to specified output directory.
```

Comment:
I changed the behavior to rewrite the encrypted file in place after moving it a specified directory (in the context of the middleware this would happen after moving the file to a new volume). I couldn't think of a reason to keep the encrypted content, since I don't retry decryption if it fails and I'm assuming that no work is being done on the encrypted version if the user is using the middleware to decrypt it.